tests/http/test_client.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from httpx import ConnectTimeout, MockTransport, Request, Response, codes
+
+from mpt_api_client.http.client import MPTClient
+
+API_TOKEN = "test-token"
+API_URL = "https://api.example.com"
+
+
+@pytest.fixture
+def mock_transport() -> MockTransport:
+    def handler_request(request: Request):  # noqa: WPS430
+        if request.url.path == "/":
+            return Response(codes.OK, json={"message": "Hello, World!"})
+        if request.url.path == "/timeout":
+            raise ConnectTimeout("Mock Timeout")
+        return Response(codes.NOT_FOUND, json={"message": "Not Found"})
+
+    return MockTransport(handler=handler_request)
+
+
+@pytest.fixture
+def mock_api_client(mocker, mock_transport: MockTransport):
+    transport_mock = mocker.patch("mpt_api_client.http.client.httpx.HTTPTransport")
+    transport_mock.return_value = mock_transport
+    return MPTClient(base_url=API_URL, api_token=API_TOKEN)
+
+
+def test_mpt_client_initialization():
+    client = MPTClient(base_url=API_URL, api_token=API_TOKEN)
+
+    assert client.api_token == API_TOKEN
+    assert client.base_url == API_URL
+
+
+def test_mpt_client_headers():
+    client = MPTClient(base_url=API_URL, api_token=API_TOKEN)
+
+    assert client.headers["Authorization"] == "Bearer test-token"
+    assert client.headers["User-Agent"] == "swo-marketplace-client/1.0"
+
+
+def test_mpt_client_timeout_and_retries():
+    client = MPTClient(
+        base_url=API_URL,
+        api_token=API_TOKEN,
+        timeout=12,  # noqa: WPS432
+        retries=2,  # noqa: WPS432
+    )
+
+    assert client._timeout.connect == 12  # noqa: WPS432, SLF001
```

Comment:
And with SLF001 I totally agree, if you need to access protected/private data to do proper tests either your tests are incorrect or your implementation is hard to test.

Reply:
most probably (my opinion you don't need this test at all) :-)